### PR TITLE
Add babel-plugin-transform-react-remove-prop-types

### PR DESCRIPTION
**Pull request by sleepwalky**
# What:
Adds **babel-plugin-transform-react-remove-prop-types** plugin for removing unnecessary React propTypes from the production build.

image

I intentionally made an error in prop-types in 'Your searches' nav item. 
https://bitbucket.org/xivart/rwd_cheaptickets_nl/branch/FI-5-speed-up-optimize-building-and-bund#diff
https://fi-5-acceptance.vayama.com/

https://fi-5-acceptance.cheaptickets.nl/en - this version was built without 'NODE_ENV=production', so we can check that our other environments are unaffected and prop-types warnings are still displayed.
It proves that this babel-plugin works at least.

### CODE REVIEW

**Comment by sleepwalky** [comment]
Does anybody know more elegant way to apply this plugin only for production builds?

File: index.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ const babelPluginTransformClassProperties = require('babel-plugin-transform-clas
 const babelPluginTransformExportDefault = require('babel-plugin-transform-export-default');
 const babelPluginTransformEs2015ObjectSuper = require('babel-plugin-transform-es2015-object-super');
 const babelPluginTransformFunctionBind = require('babel-plugin-transform-function-bind');
+const babelPluginTransfromRemovePropTypes = require("babel-plugin-transform-react-remove-prop-types").default;
```
Comment (maximuk) [inline index.js]: Is it mandatory to import with `default`?

Comment (sleepwalky) [reply index.js]: Yeap, it's not working otherwise

Comment (sleepwalky) [reply index.js]: But I'd better double check...

**Comment by yurist38** [comment]
@sleepwalky What is an issue if it's applied to another environment build (dev/staging)?

**Comment by maximuk** [comment]
@yurist38 we won't see any warnings related to proptypes. The idea is to reduce production bundle size

File: index.js (hunk 1 above)
Comment (rbardini) [reply index.js]: I was wondering the same. What happens when you don't use `default`?

File: index.js [hunk 2]
```diff
@@ -29,5 +32,6 @@ module.exports = {
     babelPluginTransformEs2015ObjectSuper,
     babelPluginTransformExportDefault,
     babelPluginTransformFunctionBind,
-  ],
+    isProduction ? babelPluginTransfromRemovePropTypes : null,
```
Comment (rbardini) [inline index.js]: Since you are filtering the plugins by boolean, you could shorten this to
```js
isProduction && babelPluginTransfromRemovePropTypes,
```

File: index.js (hunk 1 above)
Comment (sleepwalky) [reply index.js]: Double checked. Got an exception:
```
Error: Plugin 5 specified in "/Users/aliaksandr_kleshchuk/Work/rwd_cheaptickets_nl/node_modules/babel-preset-travix/index.js" provided an invalid property of "default" (While processing preset: "/Users/aliaksandr_kleshchuk/Work/rwd_cheaptickets_nl/node_modules/babel-preset-travix/index.js")
```
Probably that's why: https://github.com/loganfsmyth/babel-plugin-transform-builtin-extend/issues/1#issuecomment-172299289

File: index.js (hunk 2 above)
Comment (sleepwalky) [reply index.js]: indeed, thx
